falcon/util/structures.py

```diff
@@ -107,6 +107,126 @@ def __repr__(self):
+    also implements mapping interface, that is, object attributes are linked to
+    dictionary items, and vice versa, for instance:
+
+    >>> context = falcon.Context()
+    >>> context.cache_strategy = 'lru'
+    >>> context.get('cache_strategy')
+    'lru'
+    >>> 'cache_strategy' in context
+    True
+
+    Note:
+        Python 2 specific ``dict`` methods are exposed regardless of the
+        Python language version, however, as they are delegated to the
+        underlying ``__dict__``, a similar error would be raised as if
+        attempting to use these methods for a usual Python 3 dict.
+    """
+
+    def __contains__(self, key):
+        return self.__dict__.__contains__(key)
+
+    def __delitem__(self, key):
+        self.__dict__.__delitem__(key)
+
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self.__dict__.__eq__(other.__dict__)
+        return self.__dict__.__eq__(other)
+
+    def __getitem__(self, key):
+        return self.__dict__.__getitem__(key)
+
+    def __hash__(self):
+        return hash(self.__dict__)
+
+    def __iter__(self):
+        return self.__dict__.__iter__()
+
+    def __len__(self):
+        return self.__dict__.__len__()
+
+    def __ne__(self, other):
+        if isinstance(other, type(self)):
+            return self.__dict__.__ne__(other.__dict__)
+        return self.__dict__.__ne__(other)
+
+    def __repr__(self):
+        return '{}({})'.format(type(self).__name__, self.__dict__.__repr__())
+
+    def __setitem__(self, key, value):
+        return self.__dict__.__setitem__(key, value)
+
+    def __str__(self):
+        return '{}({})'.format(type(self).__name__, self.__dict__.__str__())
+
+    def clear(self):
+        return self.__dict__.clear()
+
+    def copy(self):
+        ctx = type(self)()
+        ctx.update(self.__dict__)
+        return ctx
+
+    def get(self, key, default=None):
+        return self.__dict__.get(key, default)
+
+    def has_key(self, key):
+        return self.__dict__.has_key(key)  # noqa
+
+    def items(self):
+        return self.__dict__.items()
+
+    def iteritems(self):
+        return self.__dict__.iteritems()
+
+    def iterkeys(self):
+        return self.__dict__.iterkeys()
+
+    def itervalues(self):
+        return self.__dict__.itervalues()
+
+    def keys(self):
+        return self.__dict__.keys()
+
+    def pop(self, key, default=None):
+        return self.__dict__.pop(key, default)
+
+    def popitem(self):
+        return self.__dict__.popitem()
+
+    def setdefault(self, key, default_value=None):
+        return self.__dict__.setdefault(key, default_value)
+
+    def update(self, items):
+        self.__dict__.update(items)
+
+    def values(self):
+        return self.__dict__.values()
+
+    def viewitems(self):
+        return self.__dict__.viewitems()
+
+    def viewkeys(self):
+        return self.__dict__.viewkeys()
+
+    def viewvalues(self):
+        return self.__dict__.viewvalues()
+
```

Comment:
Looks like all the mapping interface methods are here. 👍

Reply:
Almost :slightly_smiling_face: I shied away from some purely syntactical ones, like `__format__`, since it doesn't seem to affect much.